ovos_core/skill_manager.py

```
@@ -588,19 +589,20 @@ def _unload_plugin_skill(self, skill_id: str) -> None:
-                    skill_loader.instance.default_shutdown()
-                except Exception:
-                    LOG.exception('Failed to shutdown skill: ' + skill_loader.skill_id)
-            self.plugin_skills.pop(skill_id)
+        with self._plugin_skills_lock:
+            if skill_id in self.plugin_skills:
+                LOG.info('Unloading plugin skill: ' + skill_id)
+                skill_loader = self.plugin_skills[skill_id]
+                if skill_loader.instance is not None:
+                    try:
+                        skill_loader.instance.shutdown()
+                    except Exception:
```

Ruff (0.15.5)

[warning] Do not catch blind exception: `Exception`

(BLE001)